src/zope/app/apidoc/codemodule/class_.py

```diff
@@ -74,17 +70,33 @@ def _iterAllAttributes(self):
                     name, self.__all_ifaces, asPath=False)
             yield name, getattr(self.__klass, name), iface
 
-    def getAttributes(self):
-        """See IClassDocumentation."""
-        return [(name, obj, iface)
-                for name, obj, iface in self._iterAllAttributes()
-                if not (ismethod(obj) or ismethoddescriptor(obj))]
+    if str is bytes:
+        # Python 2
+        def getAttributes(self):
+            """See IClassDocumentation."""
+            return [(name, obj, iface)
+                    for name, obj, iface in self._iterAllAttributes()
+                    if not (ismethod(obj) or ismethoddescriptor(obj))]
+
+        def getMethods(self):
+            """See IClassDocumentation."""
+            return [(name, obj, iface)
+                    for name, obj, iface in self._iterAllAttributes()
+                    if ismethod(obj)]
+    else:
+        # XXX: This may not really be exactly correct.
+        def getAttributes(self):
+            """See IClassDocumentation."""
+            return [(name, obj, iface)
+                    for name, obj, iface in self._iterAllAttributes()
+                    if not callable(obj)]
+
+        def getMethods(self):
+            """See IClassDocumentation."""
+            return [(name, obj, iface)
+                    for name, obj, iface in self._iterAllAttributes()
+                    if callable(obj)]
 
-    def getMethods(self):
-        """See IClassDocumentation."""
-        return [(name, obj, iface)
-                for name, obj, iface in self._iterAllAttributes()
-                if ismethod(obj)]
```

Comment:
Let's keep the same getMethods/getAttributes() implementations and instead introduce a `self._ismethod()`.

Hm.  Actually I see there are three groups: methods, method descriptors, and attributes.

What are "method descriptors"?